### PR TITLE
update to kodi 19

### PR DESCRIPTION
**Pull request by outsideset**
python3, tweak APIs

### CODE REVIEW

**Comment by elbowz** [comment]
Thanks for your work @outsideset ! :) 

I cannot test the addon, due the fact I don't have kodi 19 and on my distribution (arch linux) seems I have to compile it...but I haven't free time in this period.  

So if you have tested, and if for you it's ok, I can merge and push in the matrix official repository

...or wait some days the kodi 19 became available and diffused

edit: I able to run kodi 19 and make some test

File: PartymodeAutostart.py [hunk 1]
```diff
@@ -12,11 +12,8 @@
 
 
 def log(txt):
-    if isinstance(txt, str):
-        txt = txt.decode("utf-8")
```
Comment (elbowz) [inline PartymodeAutostart.py:16]: Why have you removed the utf-8 encode/decode?

File: PartymodeAutostart.py [hunk 2]
```diff
@@ -35,8 +32,11 @@ def __init__(self):
         if self.startupPartyMode and self._viewCountdown():
             self.runPartyMode()
 
-        while not xbmc.abortRequested:
-            xbmc.sleep(1000)
+        KODIMONITOR = xbmc.Monitor()
+        KODIMONITOR.waitForAbort( 5 )            
+
+        # while not xbmc.abortRequested:
+        #     xbmc.sleep(1000)
```
Comment (elbowz) [inline PartymodeAutostart.py:39]: I'm not sure do the same thing

File: PartymodeAutostart.py [hunk 3]
```diff
@@ -151,7 +151,7 @@ def _viewCountdown(self):
 
             seconds = str((startDelayPartyModeSeconds - time) / 1000) + ' seconds'
 
-            conutdownDlg.update(percent, '', '', seconds)
+            conutdownDlg.update(percent)
```
Comment (elbowz) [inline PartymodeAutostart.py:154]: There is a reason why you have removed the seconds left?
`xbmcgui.DialogProgress().update(percent[, message])`

**Review by elbowz — COMMENTED** [review]
I have put some comments in your source only to understand....

...and I have made some rapid test, and there are some issues:
* ~*"Run Favorites"* doesn't work~ It works...my bad
* *"Run on Screensaver"* doesn't work

File: PartymodeAutostart.py (hunk 1 above)
Comment (outsideset) [reply PartymodeAutostart.py:16]: old code doesn't work in python3. see https://www.pythoncentral.io/encoding-and-decoding-strings-in-python-3-x/

File: PartymodeAutostart.py (hunk 2 above)
Comment (outsideset) [reply PartymodeAutostart.py:39]: the kodi api changed. fix taken from here https://forum.kodi.tv/showthread.php?tid=354577

File: PartymodeAutostart.py (hunk 3 above)
Comment (outsideset) [reply PartymodeAutostart.py:154]: kodi api changed. see https://codedocs.xyz/AlwinEsch/kodi/group__python___dialog_progress.html#gaa30bc06049eb516987569ae65f857cff

File: PartymodeAutostart.py (hunk 1 above)
Comment (elbowz) [reply PartymodeAutostart.py:16]: Great!

File: PartymodeAutostart.py (hunk 2 above)
Comment (elbowz) [reply PartymodeAutostart.py:39]: Yeah, I see...but they do different things.

The previous code keep the addon alive (ie. like a daemon), your code wait 5 seconds (or an abort message) and close the addon.

File: PartymodeAutostart.py (hunk 3 above)
Comment (elbowz) [reply PartymodeAutostart.py:154]: The method signature is changed, but we can always print the reaming seconds :)

**Comment by elbowz** [comment]
I have locally fixed some bugs/regressions.
If for you it's ok, I'll fix in the master appling the fix on it.

I don't know if I can commit in this PR, I guess missing rights in your fork...

**Comment by outsideset** [comment]
I recommend:
- merge my repository/master branch into a new feature branch in your repository
- add your local changes to the new feature branch
- test/iterate
- merge to your repository master branch when ready

But I'm open to doing it however is convenient. I don't have a lot of time for this project.

**Comment by elbowz** [comment]
If for "my repository/master" you mean the #13, it's what I meant.

You have already done enough 👍

**Pull request merged**